### ISSUE TITLE
Delete table entities inside transaction not working.

### ISSUE THREAD

**Issue by MidasLamb** [Azure/azure-sdk-for-rust#364]
When trying to delete entities from a table in a transaction, it returns `AtomFormatNotSupported`. I already noticed that the `DELETE` inside the multipart form has a body (being the json of the entity). But even after modifying the code that it doesn't include any body with the delete request, it still returns said error.

Small excerpt to clarify the issue:
```rust
// Deleting
let mut transaction = Transaction::default();

transaction.add(
    entity_client_a
        .delete()
        .to_transaction_operation(&entity_a)?,
);
transaction.add(
    entity_client_b
        .delete()
        .to_transaction_operation(&entity_b)?,
);

let response = partition_key_client
    .submit_transaction()
    .execute(&transaction)
    .await?
```

which prints that it's going to send this request:
```bash
--changeset_c503ee0f-6b94-49b7-9a0e-11d120bbbc1e
Content-Type: application/http
Content-Transfer-Encoding: binary

DELETE https://samlatestafgql.table.core.windows.net/aaa(PartitionKey='city',RowKey='surnameA') HTTP/1.1

--changeset_c503ee0f-6b94-49b7-9a0e-11d120bbbc1e
Content-Type: application/http
Content-Transfer-Encoding: binary

DELETE https://samlatestafgql.table.core.windows.net/aaa(PartitionKey='city',RowKey='surnameB') HTTP/1.1


--changeset_c503ee0f-6b94-49b7-9a0e-11d120bbbc1e--
--batch_93c6c5cd-18fb-4f0b-b57f-28941fb209e6
```
with corresponding response:

```bash
--batchresponse_b4e31c23-5bf3-478a-b0c7-52c3671da773
Content-Type: multipart/mixed; boundary=changesetresponse_6ebd6469-a9ba-439c-9ec8-8e5c37dd0d6a

--changesetresponse_6ebd6469-a9ba-439c-9ec8-8e5c37dd0d6a
Content-Type: application/http
Content-Transfer-Encoding: binary

HTTP/1.1 415 Unsupported Media Type
X-Content-Type-Options: nosniff
DataServiceVersion: 1.0;
Content-Type: application/xml;charset=utf-8

<?xml version="1.0" encoding="utf-8"?><error xmlns="http://schemas.microsoft.com/ado/2007/08/dataservices/metadata"><code>AtomFormatNotSupported</code><message xml:lang="en-US">Atom format is not supported.
RequestId:031cf0a9-7002-00b1-424e-ae3811000000
Time:2021-09-20T18:36:14.7162371Z</message></error>
--changesetresponse_6ebd6469-a9ba-439c-9ec8-8e5c37dd0d6a--
--batchresponse_b4e31c23-5bf3-478a-b0c7-52c3671da773--
```

I'm more than happy to help with a PR, but I need some guidance because I don't see what's wrong with the request, I've tried a few things (e.g. change content-type to json) but nothing seems to work.

**Comment by MidasLamb** [comment]
After some more investigation, and comparing with azure-sdk-for-net, it seems that the .NET SDK was sending more headers, of which it seems that the uncommented once are the required ones:
```rust
let request = request.header("Accept", "application/json;odata=minimalmetadata");
// let request = request.header("Prefer", "return-no-content");
// let request = request.header("DataServiceVersion", "3.0");
let request = request.header("If-Match", "*");
//let request = request.header("x-ms-version", "2019-02-02");
```

**Comment by MidasLamb** [comment]
Fixed by #367
